frontend/src/metabase/lib/expressions/diagnostics.js

```diff
@@ -34,7 +34,7 @@ export function countMatchingParentheses(tokens) {
   return tokens.reduce(count, 0);
 }
 
-export function diagnose(source, startRule, query) {
+export function diagnose(source, startRule, query, name = null) {
```

Comment:
This inconsistency with how we pass arguments is a bit disconcerting. Maybe we should move the arguments into a single object arg?

Reply:
The idea is to convey that `name` is indeed optional (e.g. there is no associated name for the custom expression for the filter, but there is one for a custom column).
What's a good alternative?